platform-core-module/src/main/java/com/softicar/platform/core/module/program/QueuedProgramExecutionDaemon.java

```diff
@@ -102,6 +116,29 @@ private void terminate(Collection<IRunnableThread<ProgramExecutionRunnable>> run
 		}
 	}
 
+	private void checkMaximumRuntime(AGProgram program, AGProgramExecution currentExecution) {
```

Comment:
I think this should be renamed to `terminateProgramIfRuntimeExceeded`... the current name seems to imply a mere (read-only) check.

Reply:
that name would also be wrong, since it doesnt always termiate
It always sets the exceeded flag and creates an event